internal/app/machined/pkg/runtime/v1alpha1/platform/platform.go

```diff
@@ -23,6 +23,21 @@ import (
 	"github.com/talos-systems/talos/pkg/machinery/constants"
 )
 
+// ErrNoConfigSource indicates that the platform does not have a configured source for the configuration.
+var ErrNoConfigSource = errors.New("no configuration source")
+
+// NoConfigSourceError defines an error interface which indicates the absence of a configuration source reference.
+type NoConfigSourceError interface {
+	NoConfigurationSource()
+}
+
+// IsNoConfigSourceError indicates whether the given error contains a No Configuration Source defined error.
+func IsNoConfigSourceError(err error) bool {
+	_, ok := err.(NoConfigSourceError)
+
+	return ok
+}
+
```

Comment:
We should be able to skip this and just use `errors.Is` no? Then `container` and `metal` platforms can simply return `platform.ErrNoConfigSource`.

Reply:
The whole reason I have this roundabout interface is because we can't import `platform` from the platform implementations, because `platform` imports each of these.